File: tools/status.js
#!/usr/bin/env node
// Show live server standings per graded problem: the best (lowest) score on the
// board, how many teams fully solved it, and the true total case count (public +
// private, which the per-problem API masks). API-only, no auth needed.
//   node tools/status.js
const L = require('./lib.js');

(async () => {
  const probs = await L.listProblems();
  const graded = probs.filter(p => p.status !== 'practice')
    .sort((a, b) => (a.problemSetName || '').localeCompare(b.problemSetName || '') || (a.orderInSet || 0) - (b.orderInSet || 0));
  console.log('problem                     set          board-best   solvers   cases(pub→total)');
  console.log('-'.repeat(84));
  for (const p of graded) {
    const st = await L.problemStandings(p.id);
    let best = '—', solvers = 0, total = '?';
    if (st && Array.isArray(st.rows)) {
      const full = st.rows.filter(r => r.rank != null && r.score != null);
      solvers = full.length;
      if (full.length) best = Math.min(...full.map(r => r.score));
      if (st.rows.length) total = Math.max(...st.rows.map(r => r.casesTotal || 0));
    }
    const pub = '?'; // public count needs a per-problem fetch; omitted for speed
    console.log(
      `${(p.name || p.slug).padEnd(27)} ${(p.problemSetName || '').padEnd(12)} ` +
      `${String(best).padStart(9)}   ${String(solvers).padStart(5)}    →${total}`
    );
  }
  console.log('\n(best = lowest footprint²×ticks among full-solvers; total = public+private case count)');
  process.exit(0);
})().catch(e => { console.error(e); process.exit(1); });

File: tools/status.py
#!/usr/bin/env python3
"""Live server standings per graded problem: best (lowest) score on the board, how
many teams fully solved it, and the true total case count (public + private, which
the per-problem API masks). API-only, no auth.

  python3 tools/status.py
"""
import lib


def main():
    probs = lib.list_problems()
    graded = [p for p in probs if p.get("status") != "practice"]
    graded.sort(key=lambda p: (p.get("problemSetName") or "", p.get("orderInSet") or 0))
    print(f"{'problem':27} {'set':12} {'board-best':>12} {'solvers':>8} {'cases':>7}")
    print("-" * 74)
    for p in graded:
        st = lib.problem_standings(p["id"])
        best, solvers, total = "-", 0, "?"
        if st and isinstance(st.get("rows"), list):
            # full passers only — the board's cheapest scores are partial-passers (see ours.py)
            full = [r for r in st["rows"] if r.get("rank") is not None and r.get("score") is not None
                    and r.get("casesPassed") == r.get("casesTotal")]
            solvers = len(full)
            if full:
                best = round(min(r["score"] for r in full), 2)
            if st["rows"]:
                total = max((r.get("casesTotal") or 0) for r in st["rows"])
        print(f"{(p.get('name') or p['slug']):27} {(p.get('problemSetName') or ''):12} {str(best):>12} {solvers:>8} {'->'+str(total):>7}")
    print("\n(best = lowest footprint^2 x ticks among full-solvers; cases = public+private total)")


if __name__ == "__main__":
    main()
